src/include/btree_inline.h

```diff
@@ -1651,8 +1664,8 @@ __wt_page_can_evict(WT_SESSION_IMPL *session, WT_REF *ref, bool *inmem_splitp)
     page = ref->page;
     mod = page->modify;
 
-    /* A truncated page can't be evicted until the truncate completes. */
-    if (__wt_page_del_active(session, ref, true))
+    /* Don't attempt to evict fast-truncate pages until the truncate completes. */
+    if (ref->page_del != NULL && ref->page_del->resolved == 0)
```

Comment:
Shouldn't we still call `__wt_page_del_active` here and instead add the additional check for resolved into that function?

Reply:
I guess the concern is that other locations call `__wt_page_del_active` for different reasons.

Reply:
Yes -- `__wt_page_del_active` is a visibility test, and I don't think we care about visibility here. I think the original assumption was if the truncate wasn't visible, there was no point in trying for eviction, but that's no longer the case, so we let reconciliation own the visibility problem, and all we're doing here is trying to predict if eviction (or at least a scrub of the dirty data on the page), is possible.

Reply:
Do we need to update this check to be:
```
if (ref->page_del != NULL && (ref->page_del->resolved == 0 || __wt_page_del_active(session, ref, true)))
```

Since we don't actively update the `resolved` flag, we still need to check on global visibility to avoid having pages in cache that were read back in while the truncate was unresolved being stuck in cache forever?

Or am I missing the bit of this change that updates the `resolved` flag when a truncate commits/aborts (that's possible, I didn't go digging far)?

Reply:
We actively update the `resolved` flag: see `__wt_delete_page_rollback()` (abort), and `__wt_txn_commit()` (commit).

Reply:
I see - thanks for explaining. In that case I agree this change looks right.